app/renderer/actions/Inspector.js

```diff
@@ -130,17 +130,22 @@ export function selectElement (path) {
     const [optimalStrategy, optimalSelector] = strategyMap.length > 0 ? strategyMap[strategyMap.length - 1] : ['xpath', selectedElementXPath];
 
     // Get the information about the element
-    const {elementId, variableName, variableType} = await callClientMethod({
+    let {elementId, variableName, variableType} = await callClientMethod({
       strategy: optimalStrategy,
       selector: optimalSelector,
     });
 
-    // Set the elementId, variableName and variableType for the selected element 
-    // (check first that the selectedElementPath didn't change, to avoid race conditions)
-    if (getState().inspector.selectedElementPath === path) {
-      dispatch({type: SET_SELECTED_ELEMENT_ID, elementId, variableName, variableType});
-    }
-    
+    if (!elementId) {
+      // If the element is stale, unselect and reload the source
+      dispatch({type: UNSELECT_ELEMENT});
+      await applyClientMethod({methodName: 'source'})(dispatch);
```

Comment:
So we reset the selection for this case?

Reply:
Yeah. Because the element is stale, we dispatch UNSELECT_ELEMENT so that there isn't a selected element anymore.

Reply:
Isn't it possible to restore the selection after reload happens?